extensions/stackdriver/metric/record.cc

```diff
@@ -60,54 +56,185 @@ void record(bool is_outbound, const ::Wasm::Common::FlatNode& local_node_info,
 
   const auto peer_rev_iter =
       peer_labels ? peer_labels->LookupByKey(
                         Wasm::Common::kCanonicalServiceRevisionLabelName.data())
                   : nullptr;
   const auto peer_canonical_rev =
       peer_rev_iter ? peer_rev_iter->value() : nullptr;
+  return peer_canonical_rev ? peer_canonical_rev->str()
+                            : ::Wasm::Common::kLatest.data();
+}
 
-  if (is_outbound) {
-    opencensus::tags::TagMap tagMap = {
-        {meshUIDKey(), flatbuffers::GetString(local_node_info.mesh_id())},
-        {requestOperationKey(), operation},
-        {requestProtocolKey(), request_info.request_protocol},
-        {serviceAuthenticationPolicyKey(),
-         ::Wasm::Common::AuthenticationPolicyString(
-             request_info.service_auth_policy)},
-        {destinationServiceNameKey(), request_info.destination_service_name},
-        {destinationServiceNamespaceKey(),
-         flatbuffers::GetString(peer_node_info.namespace_())},
-        {destinationPortKey(), std::to_string(request_info.destination_port)},
-        {responseCodeKey(), std::to_string(request_info.response_code)},
-        {sourcePrincipalKey(), request_info.source_principal},
-        {sourceWorkloadNameKey(),
-         flatbuffers::GetString(local_node_info.workload_name())},
-        {sourceWorkloadNamespaceKey(),
-         flatbuffers::GetString(local_node_info.namespace_())},
-        {sourceOwnerKey(), Common::getOwner(local_node_info)},
-        {destinationPrincipalKey(), request_info.destination_principal},
-        {destinationWorkloadNameKey(),
-         flatbuffers::GetString(peer_node_info.workload_name())},
-        {destinationWorkloadNamespaceKey(),
-         flatbuffers::GetString(peer_node_info.namespace_())},
-        {destinationOwnerKey(), Common::getOwner(peer_node_info)},
-        {destinationCanonicalServiceNameKey(),
-         flatbuffers::GetString(peer_canonical_name)},
-        {destinationCanonicalServiceNamespaceKey(),
-         flatbuffers::GetString(peer_node_info.namespace_())},
-        {destinationCanonicalRevisionKey(),
-         peer_canonical_rev ? peer_canonical_rev->str()
-                            : ::Wasm::Common::kLatest.data()},
-        {sourceCanonicalServiceNameKey(),
-         flatbuffers::GetString(local_canonical_name)},
-        {sourceCanonicalServiceNamespaceKey(),
-         flatbuffers::GetString(local_node_info.namespace_())},
-        {sourceCanonicalRevisionKey(), local_canonical_rev
-                                           ? local_canonical_rev->str()
-                                           : ::Wasm::Common::kLatest.data()}};
+TagKeyValueList getOutboundTagMap(
+    const ::Wasm::Common::FlatNode& local_node_info,
+    const ::Wasm::Common::FlatNode& peer_node_info,
+    const ::Wasm::Common::RequestInfo& request_info) {
+  TagKeyValueList outboundMap = {
+      {std::make_pair<opencensus::tags::TagKey, std::string>(
```

Comment:
I am wondering do you need to add `std::make_pair<opencensus::tags::TagKey, std::string>`, or just the the pair of strings would also compile, like the original code.

Reply:
pair of strings worked... it didn't work originally because I think there were some stringview in there and I had to convert them to std::string.. fixed